source/Halibut.Tests/Support/PendingRequestQueueFactories/CancelWhenRequestDequeuedPendingRequestQueueFactory.cs

```diff
@@ -67,6 +67,11 @@ public async Task ApplyResponse(ResponseMessage response, ServiceEndPoint destin
 
             public async Task<ResponseMessage> QueueAndWaitAsync(RequestMessage request, CancellationToken requestCancellationToken)
                 => await inner.QueueAndWaitAsync(request, requestCancellationToken);
+
+            public ValueTask DisposeAsync()
```

Comment:
Ooh, nice one. 

Just wondering, how did this manifest as a problem?

Reply:
One of the tests would enqueue work, deque that work then never respond. I saw tests never finishing dumped process and could see the stack trace.

I think before with the timeout it would eventually undo.

Without the timeout removed here that would never happen. Note that when used for real we ALWAYS respond to the in memory queue.